modules/git/tree_entry.go

```diff
@@ -179,3 +181,30 @@ func (tes Entries) Sort() {
 func (tes Entries) CustomSort(cmp func(s1, s2 string) bool) {
 	sort.Sort(customSortableEntries{cmp, tes})
 }
+
+// GetPathInRepo returns the relative path in the tree to this entry
+func (te *TreeEntry) GetPathInRepo() string {
+	if te == nil {
+		return ""
+	}
+
+	path := te.Name()
+	current := te.ptree
+
+	for current != nil && current.ptree != nil {
+		entries, err := current.ptree.ListEntries()
+		if err != nil {
+			log.Error("Failed to climb git tree %v", err)
+			return ""
+		}
+		for _, entry := range entries {
+			if entry.ID.String() == current.ID.String() {
+				path = entry.Name() + "/" + path
+				break
+			}
+		}
+		current = current.ptree
+	}
```

Comment:
This looks like a really inefficient way to gather the full path of a file.
However, I know that this function is drastically needed, also for other functionality (see for example #28835 which got stalled exactly as I wasn't sure how to implement this)._
Thus, I'll allow it for now as long as we get it merged.
We can still improve its performance in the future with a precalculated path.

Reply:
Agreed. I think a TreeEntry should have its absolute path. But, felt out of scope for this requested feature.